portfolio/src/main/java/com/google/sps/servlets/CommentServlet.java

```diff
@@ -36,19 +37,30 @@
 
   private List<String> messages;
 
+  public static LinkedHashMap<String, Integer> commentsByPerson = new LinkedHashMap<>();
```

Comment:
```suggestion
 static Map<String, Integer> commentsByPerson = new LinkedHashMap<>();
```

Package private seems enough for commentsByPerson, we don't give access of class variables or methods unless it is necessary. Sometimes, even have getters or setters for private variables to protect them. https://docs.oracle.com/javase/tutorial/java/javaOO/accesscontrol.html.

When declare a variable, we usually define it as the generic type, Map vs LinkedHashMap in this case. So that if there a method that takes only HashMap, this variable can be passed in without converting the type. Ideally, the methods parameters should also be generic types as well, so that it can be more flexible. 

Another question, is any feature of LinkedHashMap used? e.g. can commentsByPerson just be a HashMap?

Reply:
I have changed  he static variable to be Package private and looked and the extra documentation that you sent, thank you.
 
I fixed the type to be Generic as Map as well. 

The reason that I choose a LinkedHashMap rather than HashMap is because of the kept insertion order of the comments in case I wanted to change the type of graph so that I would be able to view the number of comments overtime. Other than that I believe I could use a normal HashMap. Would you suggest that I change that to a HashMap until I utilize its features more?

Reply:
I would recommend use HashMap for now, and when you really depend on features of LinkedHashMap, then update the type.